src/agent/sqlite_manager/src/sqlite_manager.cpp

```diff
@@ -0,0 +1,210 @@
+        for (const Col& col : cols)
+        {
+            queryString += col.m_name + " " + mapColTypeString.at(col.m_type);
+            if (col.m_primaryKey)
+                queryString += " PRIMARY KEY";
+            if (col.m_autoIncrement)
+                queryString += " AUTOINCREMENT";
+            if (col.m_notNull)
+                queryString += " NOT NULL";
+            queryString += ", ";
+        }
+        // Remove the trailing ", "
+        queryString.erase(queryString.size() - 2);
+        queryString += ");";
+
+        std::cout << "QueryString: " + queryString << std::endl;
+
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error initializing table: " << e.what() << std::endl;
+            throw;
+        }
+        return 1;
+    }
+
+    int SQLiteManager::Insert(const std::string& tableName, const std::vector<Col>& cols)
+    {
+        std::string queryString = "INSERT INTO " + tableName + " (";
+
+        for (const Col& col : cols)
+        {
+            queryString += col.m_name + ",";
+        }
+        // Remove the trailing ", "
+        queryString.erase(queryString.size() - 1);
+        queryString += ") VALUES (";
+
+        for (const Col& col : cols)
+        {
+            if (col.m_type == SQLite::TEXT)
+                queryString += "\"";
+            queryString += col.m_value;
+            if (col.m_type == SQLite::TEXT)
+                queryString += "\"";
+            queryString += ",";
+        }
+        // Remove the trailing ", "
+        queryString.erase(queryString.size() - 1);
+        queryString += ");";
+
+        std::cout << "QueryString: " + queryString << std::endl;
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error inserting element: " << e.what() << std::endl;
+            throw;
+        }
+        return 1;
+    }
+
+    int SQLiteManager::GetCount(const std::string& tableName)
+    {
+        std::string queryString = std::format("SELECT COUNT(*) FROM {}", tableName);
+
+        std::cout << "QueryString: " + queryString << std::endl;
+        try
+        {
+            SQLite::Statement query(*m_db, queryString);
+            int count = 0;
+            if (query.executeStep())
+            {
+                count = query.getColumn(0).getInt();
+            }
+            else
+            {
+                std::cerr << "Error getting element count." << std::endl;
+            }
+            return count;
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error during GetCount operation: " << e.what() << std::endl;
+            return {};
+        }
+    }
+
+    int SQLiteManager::ExecuteSQL(const std::string& queryString)
```

Comment:
I think we should clarify in the name of the function that this won't return values, for example, for `select` queries. Besides, a return value won't be necessary.

My suggestion:
```suggestion
    void SQLiteManager::ExecuteNoSelectSQL(const std::string& queryString)
```

Reply:
I agree. Let me hold on to this for a bit until I write the Remove() function.

Reply:
Done.